tools/prepare_job.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""This script is designed to generate a Job-file in yaml, which will
+be used with our RevPi devices in order to flash and perform tests with
+LAVA automatically, thus saving time finding errors in the images.
+"""
+
+import argparse
+import os
+import sys
+import yaml
+
+NEW_JOB_NAME = "job_flash_test.yaml"
+REVPI_DEFAULT_DEVICE = "RevPi_Core"
+PATH_DEPLOYMENT_FLASH = "./automated/deployment/"
+PATH_TESTSUITES = "./automated/testsuites/"
+
+parser = argparse.ArgumentParser(
+    description=__doc__,
+    epilog="Default behaviour: blablabla...",
+)
+
+parser.add_argument(
+    "-d",
+    "--device_type",
+    dest="device_type",
+    nargs="?",
+    default=REVPI_DEFAULT_DEVICE,
+    help="device type ...",
+)
+
+parser.add_argument(
+    "-u",
+    "--url",
+    dest="url",
+    required=True,
+    help="url ...",
+)
+
+parser.add_argument(
+    "-f",
+    "--flash",
+    dest="flash",
+    required=True,
+    help="flash ...",
+)
+
+parser.add_argument(
+    "-s",
+    "--testsuite",
+    dest="testsuite",
+    required=True,
+    help="testsuite ...",
+)
+
+args = parser.parse_args()
+
+#TODO what about another URLs???
+def get_imagename_from_url(url):
+    return os.path.basename(url)
+
+def add_namespace_in_testsuite(content):
+    actions = content.get("actions", [])
+
+    for action in actions:
+        if "test" in action or "boot" in action or "deploy" in action:
+            action_name = next(iter(action))
+            action[action_name]["namespace"] = "testsuite"
+
+    return content
+
+def create_job(device_type, image_url, file_flash, file_testsuite) -> bool:
+    print("Generate Job... ", end="")
+    
+    # Load the contents of File 1
+    try:
+        with open(file_flash, 'r') as file:
+            file1_contents = yaml.safe_load(file)
```

Comment:
A more specific name than `file1_contents` would be nice - maybe something like `file_flash_content`?